File: https:/github.com/rajesh-kayal-dev/Leetcode.git/347-TopKFrequentElements/347-TopKFrequentElements.java
// Last updated: 6/15/2026, 10:00:39 AM
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> map =  new HashMap<>();

        for(int num : nums){
            map.put(num, map.getOrDefault(num, 0)+1);
            //[1,1,2,3,1] → freqMap: {1=3, 2=1, 3=1}
        }
        PriorityQueue<Map.Entry<Integer,Integer>> maxHeap = 
        new PriorityQueue<>((a,b)-> b.getValue() - a.getValue());

        //Max Heap (PriorityQueue)— based on frequency:
        //Highest frequency sabse pehle aaye
        //Custom comparator  b.getValue() - a.getValue() ke through.

        maxHeap.addAll(map.entrySet());
        //heap mein sab elements sorted hain by highest frequency

        int result[] = new int[k];
        // result array banaya gaya hai jisme hum top k elements store kareng

        for(int i =0; i < k; i++){
            result[i] = maxHeap.poll().getKey();//poll() → heap ka top element nikalta hai (highest freq)
        }
        return result;
    }
}

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/347-TopKFrequentElements/347-TopKFrequentElements.js
// Last updated: 7/25/2026, 10:14:31 AM
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
   //count frequescy
   //sort decending order
   //return top k
    let count = {};

    for(let num of nums){
        count[num] = (count[num] || 0 ) + 1; //count frequency of word
    }

   return Object.entries(count) //coveting object into array for sort operation
            .sort((a,b)=> b[1] - a[1]) //sort decending order
            .slice(0, k) //for pick top k values 
            .map((item) => Number(item[0])) //return result into  array of value
};